src/ObjectUtils/PdbFileLlvm.cpp

```diff
@@ -36,20 +47,72 @@ class SymbolInfoVisitor : public llvm::codeview::SymbolVisitorCallbacks {
+
+    // The ProcSym's name does not contain an argument list. However, this information is required
+    // when dealing with overloads and it is available in the type info stream. See:
+    // https://llvm.org/docs/PDB/TpiStream.html
+    llvm::StringRef argument_list = RetrieveArgumentList(proc);
+    if (!argument_list.empty()) {
+      symbol_info.set_demangled_name(
+          absl::StrCat(symbol_info.demangled_name(), argument_list.data()));
+    }
+
     // The address in PDB files is a relative virtual address (RVA), to make the address compatible
     // with how we do the computation, we need to add both the load bias (ImageBase for COFF) and
     // the offset of the executable section.
     symbol_info.set_address(proc.CodeOffset + object_file_info_.load_bias +
                             object_file_info_.executable_segment_offset);
     symbol_info.set_size(proc.CodeSize);
+
     *(module_symbols_->add_symbol_infos()) = std::move(symbol_info);
 
     return llvm::Error::success();
   }
 
  private:
+  [[nodiscard]] llvm::StringRef RetrieveArgumentList(const llvm::codeview::ProcSym& proc) const {
+    llvm::codeview::LazyRandomTypeCollection& type_collection = type_info_stream_->typeCollection();
+
+    // We expect function types being either LF_PROCEDURE or LF_MFUNCTION, which are non-simple
+    // types. However, there are cases where the function type is "<no type>. In those cases, we
+    // can't retrieve the argument list.
+    if (proc.FunctionType.isSimple()) {
+      llvm::StringRef function_type = type_collection.getTypeName(proc.FunctionType);
+      ORBIT_ERROR(
+          "Unable to retrieve parameter list for function \"%s\"; The function type is \"%s\"",
+          proc.Name, function_type);
+      return "";
+    }
+
+    llvm::codeview::CVType function_type = type_info_stream_->getType(proc.FunctionType);
+    switch (function_type.kind()) {
+      case llvm::codeview::LF_PROCEDURE: {
+        llvm::codeview::ProcedureRecord procedure_record;
+        llvm::Error error =
+            llvm::codeview::TypeDeserializer::deserializeAs<llvm::codeview::ProcedureRecord>(
+                function_type, procedure_record);
+        ORBIT_CHECK(!error);
+
+        llvm::StringRef parameter_list = type_collection.getTypeName(procedure_record.ArgumentList);
+        return parameter_list;
+      }
+      case llvm::codeview::LF_MFUNCTION: {
+        llvm::codeview::MemberFunctionRecord member_function_record;
+        llvm::Error error =
+            llvm::codeview::TypeDeserializer::deserializeAs<llvm::codeview::MemberFunctionRecord>(
+                function_type, member_function_record);
+        ORBIT_CHECK(!error);
+
+        return type_collection.getTypeName(member_function_record.ArgumentList);
+      }
+      default:
+        ORBIT_UNREACHABLE();
+    }
+  }
+
   ModuleSymbols* module_symbols_;
   ObjectFileInfo object_file_info_;
+  llvm::pdb::TpiStream* type_info_stream_;
```

Comment:
This threw me off for a second.. Its not really your change, so please feel free to ignore. My pers pref: No long function definitions in the class declaration (unless necessary)

Reply:
Agree. Moved both implementations down.